kubespawner/spawner.py

```diff
@@ -2158,3 +2310,22 @@ async def load_user_options(self):
                     )
                 )
             )
+
+    async def _ensure_namespace(self):
+        ns = make_namespace(self.namespace)
+        api = self.api
+        try:
+            api.create_namespace(ns)
+        except ApiException as e:
+            if e.status != 409:
+                # It's fine if it already exists
+                self.log.exception("Failed to create namespace %s",
+                                   self.namespace)
+                raise
+
+
+class MultiNamespaceKubeSpawner(KubeSpawner):
+    """
+    MultiNamespaceKubeSpawner is the all-namespaces analog of KubeSpawner.
+    """
+    enable_user_namespaces = True
```

Comment:
By having `enable_user_namespaces` be a configurable trait, as you've defined above, I think we can skip the definition of this additional class and refer folks to:

```python
c.KubeSpawner.enable_user_namespaces = True
```

Reply:
OK.  This is clearly a matter of taste, but I'm happy to go with that.

Reply:
Done!